### ISSUE TITLE
Review of section 4

### ISSUE THREAD

**Issue by r-owen** [lsst-dm/spie-2018-dm-practices#31]
Overall excellent. Very clear and readable. A few suggestions:

4.2.2 Language Standards

"We do not require a specific C++ compiler to build the DM software, but we do require compliance with a specific version of the standard." This implies much better compatibility than we offer. I strongly suggest we reword this to reflect the reality that we only test our software on two compilers and do not claim it runs on any others:
- Some version of gcc from dev-toolset on some flavor of linux (it is probably worth explicitly stating that we use dev-toolset because otherwise readers may panic when they realize that the gcc is far newer than what RedHat offers)
- The current version of clang on macOS

I hope someday we will test with clang on linux and test on two versions of macOS (current and the next older) but we're not there yet.

4.2.3 Unit Testing

"In DM we do almost all our testing from Python using the unittest module and we test the C++ code by using the Python bindings."
That is true, but we also have a few cases where python testing of C++ code is impractical, and in those cases we use Boost Test.

"We run the tests in a multi-process mode, based on the number of cores on the system, and all code is tested with flake8 for compliance"
Not true yet, but we're moving in that direction.

4.4.1 Software Packaging

Consider not repeating "system" in: "The first type are the system prerequisites that we need to have available on the system"
perhaps "The first type are the system prerequisites..." -> "The first type are prerequisites..."?

This is a bit odd because Anaconda Python already has those packages: "we always install our own Anaconda and install a set of Python prerequisites including numpy, matplotlib, and astropy". Perhaps "we always install our own Anaconda Python, including prerequisite packages such as numpy, matplotlib, and astropy".

"and by default knows how to build...using autoconf, SCons, and make": make all three build system names monospace? Does it support cmake as well?

You never explicitly say what fourth type of package is. The description starts on the paragraph "The LSST EUPS-based software packages". Since this is so far from the first paragraph, where the four types of package are mentioned, I suggest that you summarize all four types in the first paragraph, then expand on the third and fourth type in the remaining text. For example:

"The first type are prerequisites that we need to have available on the system: this can include compilers, and build support tools such as cmake. The second type is Python and the set of standard Python packages. Although the software should be able to be built with any valid Python, for LSST development and testing we always install our own Anaconda and install a set of Python prerequisites including numpy, matplotlib, and astropy. The third type is 3rd party packages built with standard tools. The fourth type is our own packages (and forks of a few 3rd party packages) that are fully EUPS compliant and built with our tools.

We manage most 3rd party package by creating a package constisting of a compressed tar file of the upstream distribution, an optional directory of patches to be applied after the source code has been extracted, and a ups directory containing..."

I'm not sure what to make of the last paragraph: "Support packages from the Science Quality and Reliability Engineering group...". Which of the 4 catagories are you saying this fits into? It's not at all clear to me. It sounds to me as if perhaps a mix of categories 2 and 3. Or is the software standalone, rather than directly using the DM stack, and thus not managed by EUPS?

4.4.4 Future Considerations

"creating a git repository similar to the current lsst distrib metapackage, containing git submodules to the separate git repositories"
"submodules TO the separate..." reads oddly to me. Consider "...containing its packages as git submodules"

**Comment by timj** [comment]
The paper won't be out until September at the earliest so I was being aspirational with the "all code is tested with flake8" line.
